src/libraries/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj

```diff
@@ -6,6 +6,8 @@
     <NoWarn>$(NoWarn);SA1205;CA1845</NoWarn>
     <EnableTrimAnalyzer Condition="$([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) == '.NETFramework'">false</EnableTrimAnalyzer>
     <IsPackable>true</IsPackable>
+    <!-- Lifetime rules introduced in C# 11 impact scenarios in net6 framework -->
+    <LangVersion Condition="'$(TargetFramework)' == 'net6.0'">10</LangVersion>
```

Comment:
Do you happen to remember which code specifically needed this?

Reply:
This is the issue when targeting .NET 6 but permitting preview language features. The issue is the .NET 6 reference assembly doesn't have the `scoped` so the compiler fails. The problem files are `src\libraries\System.Diagnostics.DiagnosticSource\src\System\Diagnostics\Metrics\StringSequence.netcore.cs` and `src\libraries\System.Diagnostics.DiagnosticSource\src\System\Diagnostics\Metrics\ObjectSequence.netcore.cs`. I've not tried to fix this. An outstanding question here is why is a .NET 6 build needed.

Example failure:
```
StringSequence.netcore.cs(19,49): error CS8170: Struct members cannot return 'this' or other instance members by reference
StringSequence.netcore.cs(19,20): error CS8347: Cannot use a result of 'MemoryMarshal.CreateSpan<string>(ref string, int)' in thi 
s context because it may expose variables referenced by parameter 'reference' outside of their declaration scope
```

Reply:
> I've not tried to fix this.

Ok, I'll look at fixing it up subsequently.  I'd like to avoid this LangVersion specialization whenever possible. It's fine for your PR.